lib/chef/provider/windows_task.rb

```diff
@@ -89,11 +89,17 @@ def action_create
         basic_validation
         options = {}
         options["F"] = "" if new_resource.force || task_need_update?
-        options["SC"] = schedule
+        if schedule == :none
+          options["SC"] = :once
+          options["ST"] = "00:00"
+          options["SD"] = convert_user_date_to_system_date "12/12/2012"
```

Comment:
In the original issue, the reporter indicated that we could construct XML hat doesn't require this "start time in the past" trick:

https://github.com/chef/chef/issues/6200#issuecomment-324386409

Is there a reason we can't do that here?  I see that it would require a substantial change to this class since we rely on option flags for nearly everything at the moment.  I understand if that is a bit more than we want to do at the moment, but it  might be worth a comment.

Reply:
Thank you @stevendanna for your review comments.
 As per original issue comments I have changed the XML file in `def update_task_xml`. But here the code flow is such that we are creating a task first using `schtasks` command and then updating the XML template if required, for value passed such as `execution time`, `random_delay`, etc.  So here we created a scheduled task with frequency `:once` for an outdated `start date`, so that it doesn't execute. Then we update it's XML for frequency `none` in `def update_task_xml`. 
**Note**: We need to do this because `SC` is a mandatory parameter in `schtasks` command.